### DISCUSSION TITLE
timer boundary event setup

### DISCUSSION THREAD

**Discussion by bsdgithub** [flowsforapex/apex-flowsforapex#761] [Q&A] [answered]
hello,
I'm trying to use the timer boundary event with this setup:
- cicle oracle
- timer first run 000 00:00:00
- next timeer 000 00:30:00
- max run 5
I've got the **error** on save:
 ` Get Diagram failed. TypeError: t.search is not a function
    at Wi.build (flows4apex.modeler.min.js:2:2970153)
    at flows4apex.modeler.min.js:2:2973158
    at y (flows4apex.modeler.min.js:2:2693823)
    at flows4apex.modeler.min.js:2:2973090
    at y (flows4apex.modeler.min.js:2:2693823)
    at $i.parseContainments (flows4apex.modeler.min.js:2:2972945)
    at $i.build (flows4apex.modeler.min.js:2:2970915)
    at flows4apex.modeler.min.js:2:2973421
    at y (flows4apex.modeler.min.js:2:2693823)
    at flows4apex.modeler.min.js:2:2973350`

If max run is set to null, no error.
Is it the expected behavior?
Thank you,

**Comment by rallen2010** [comment]
Which release of Flows for APEX?

**Comment by bsdgithub** [comment]
23.X

**Comment by rallen2010** [comment]
I can reproduce this - so looks like a bug in the BPMN Modeler.  Reproduces in F4A 24.1 as well.  If you have a non-null value for Max Runs in a Non interrupting Timer Boundary Event of type Oracle Cycle, the modeler won't save the model & also can't 'Show Source'.
**The work-around** is to to use ISO Cycle Timer - in this case, the Duration with Repeat syntax would be `R5/PT30M`. (The doc is also unclear that you need to have a T in the period duration (you do!) so I just created a timer that fired every 30 months, as was wondering why it hadn't fired yet!!)

**Reply by bsdgithub** [reply]
Thank you @rallen2010 .
Can you share with me the setup for doing that in ISO Cycle timer?
I saw that the bug report is already opened.

**Reply by rallen2010** [reply, marked as answer]
Workaround - instead of using a timer of type `Cycle (Oracle)`, select `Cycle (ISO 8601)`.  To get a timer that repeats unto 5 times, every 30 Minutes, use this Syntax: `R5/PT30M`.   Translating,  Repeat(R) 5 Times, using a Period (P) that's a  time (T) of 30 Minutes (M).    If you forget the T before the 30M, it will assume this is a date, not a time, and give you a cycle period of 30 Months.
For more info on this, see the documentation at [flowsforapex.org ](https://flowsforapex.org/latest/timer-syntax/)

**Reply by bsdgithub** [reply]
never used it, thanks for explanations!
